src/leap/mail/imap/account.py

```diff
@@ -244,15 +249,15 @@ def select(self, name, readwrite=1):
         """
         name = normalize_mailbox(name)
 
+        if name in self._mailboxes:
+            return defer.succeed(self._mailboxes[name])
```

Comment:
when reviewing this, I realized that the status of notifications was actually broken for us.
it was complicated by the fact that we had two different places in which we instantiated a top-level Account object, so the listener set for the underlying message_collection wasn't shared.

have a look at commit https://github.com/leapcode/leap_mail/pull/231/commits/ab50358222db1d78ba8384ac299443b4c95397ee and see if this would work for your case.